### PR TITLE
NetBSD support proposal.

### PR DESCRIPTION
**Pull request by devnexen**
Thanks a lot for making a contribution to SFML! 🙂

Before you create the pull request, we ask you to check the follow boxes. (For small changes not everything needs to ticked, but the more the better!)

* [ ] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before?
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

### CODE REVIEW

File: src/SFML/Window/CMakeLists.txt [hunk 1]
```diff
@@ -124,6 +124,13 @@ elseif(SFML_OS_LINUX OR SFML_OS_FREEBSD OR SFML_OS_OPENBSD)
             ${SRCROOT}/OpenBSD/JoystickImpl.cpp
             ${SRCROOT}/OpenBSD/JoystickImpl.hpp
         )
+    elseif(SFML_OS_NETBSD)
+        set(PLATFORM_SRC
+            ${PLATFORM_SRC}
+	    ${SRCROOT}/NetBSD/JoystickImpl.cpp
+	    ${SRCROOT}/NetBSD/JoystickImpl.hpp
```
Comment (eXpl0it3r) [inline src/SFML/Window/CMakeLists.txt:131]: Wrong indentation

File: src/SFML/Window/NetBSD/JoystickImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+namespace
+{
+    std::map<unsigned int, std::string> plugged;
+    std::map<int, std::pair<int, int> > hatValueMap;
+
+    bool isJoystick(const char *name)
+    {
+        int file = ::open(name, O_RDONLY | O_NONBLOCK);
+	int id;
```
Comment (eXpl0it3r) [inline src/SFML/Window/NetBSD/JoystickImpl.cpp:54]: Use spaces instead of tabs

File: src/SFML/Window/NetBSD/JoystickImpl.cpp [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+        if (!desc)
+        {
+            ::close(file);
+            return false;
+        }
+
+	if (ioctl(file, USB_GET_REPORT_ID, &id) < 0) {
+		::close(file);
+		return false;
+	}
```
Comment (eXpl0it3r) [inline src/SFML/Window/NetBSD/JoystickImpl.cpp:70]: Use spaces instead of tabs

File: src/SFML/Window/NetBSD/JoystickImpl.cpp [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+            {
+                ::close(m_file);
+                return false;
+            }
+
+            // And the id
+	    if (ioctl(m_file, USB_GET_REPORT_ID, &m_id) < 0) {
+                ::close(m_file);
+                return false;
+	    }
```
Comment (eXpl0it3r) [inline src/SFML/Window/NetBSD/JoystickImpl.cpp:223]: Use spaces instead of tabs

File: src/SFML/Window/NetBSD/JoystickImpl.cpp [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+            // Get the report descriptor
+            m_desc = hid_get_report_desc(m_file);
+            if (!m_desc)
+            {
+                ::close(m_file);
+                return false;
+            }
+
+            // And the id
+	    if (ioctl(m_file, USB_GET_REPORT_ID, &m_id) < 0) {
```
Comment (eXpl0it3r) [inline src/SFML/Window/NetBSD/JoystickImpl.cpp:220]: `{` should be on a new line

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
I commented on some small stylistic changes.

As someone with no knowledge or experience with NetBSD, how would I go about testing this?

File: cmake/Config.cmake [hunk 6]
```diff
@@ -35,6 +35,10 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "^OpenBSD$")
     set(SFML_OS_OPENBSD 1)
     # don't use the OpenGL ES implementation on OpenBSD
     set(OPENGL_ES 0)
+elseif(CMAKE_SYSTEM_NAME MATCHES "^NetBSD$")
+	set(SFML_OS_NETBSD 1)
+	# don't use the OpenGL ES implementation on NetBSD
```
Comment (eXpl0it3r) [inline cmake/Config.cmake:40]: Missed one, sorry: Use spaces instead of tab

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

**Comment by Bromeon** [comment]
Is there any difference in behavior regarding the different BSD distributions?
If not, we might think about collapsing the config and #define constants.

**Comment by devnexen** [comment]
Mainly in term of joystick support in this case.

**Comment by Bromeon** [comment]
Basically, your PR is very similar to the [existing FreeBSD joystick implementation](https://github.com/SFML/SFML/blob/master/src/SFML/Window/FreeBSD/JoystickImpl.cpp), in small differences e.g. how to get joystick ID. Meanwhile, the [OpenBSD implementation](https://github.com/SFML/SFML/blob/master/src/SFML/Window/OpenBSD/JoystickImpl.cpp) is an empty stub.

But in this case I'm not sure if it makes things more complicated to unify common code rather than duplicating it. The only thing I could imagine is if there is some `Window/BSD` folder with shared code -- not sure if it's worth it.

By the way, do you happen to know if the same code can be used on OpenBSD?

**Comment by devnexen** [comment]
> 
> By the way, do you happen to know if the same code can be used on OpenBSD?

It might ... maybe the only difference is the ioctl lookup id, just a glance in their code tree tough.

**Comment by eXpl0it3r** [comment]
I my opinion, code duplication is okay, if it's for the sake of isolating independent contexts/modules/etc.
Maybe there's a way to make share more code between the different BSD versions, but I think that should then maybe be done as a separate PR.

**Comment by eXpl0it3r** [comment]
Thank you for the contribution! 🙂

**Pull request merged**